### PR TITLE
ADD: EIP-5489 NFT Hyperlink Extension

### PR DESCRIPTION
**Pull request by coderfengyun**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-5438.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-08-16
+requires: 165, 721
+---
+
+## Abstract
+
+This ERC proposes a new extension of NFT (non-fungible token, aka ERC-721): nft-hyperlink-extention (hNFT), which embedded hyperlinks in NFTs. With hHNFT, the owners of NFTs can authorize a URL slot to a specific address which can be either an externally-owned account (EOA) or a contract account and they are entitled to revoke that authorization as well. Address which got slot authorization, can manage the URI on the slot.
```
Comment (Pandapip1) [inline EIPS/eip-5438.md:16]:
```suggestion
This ERC proposes a new extension of NFT (non-fungible token, aka [EIP-721](./eip-721.md)) that embeds hyperlinks in NFTs, referred to as "hNFT". With hHNFT, the owners of NFTs can authorize a URL slot to a specific address which can be either an externally-owned account (EOA) or a contract account and they are entitled to revoke that authorization as well. The address which has slot authorization can manage the URI on the slot.
```

File: EIPS/eip-5438.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+### New URI format
+
+This extension tries to add a new URI format, which is prefixed with "hnft://". This URI format enables redirect or connect activity between hNFTs, it enables great imagination.
+
+
+### Interface
+
+#### IERC5438
+```
```
Comment (Pandapip1) [inline EIPS/eip-5438.md:52]:
```suggestion
```solidity
```

File: EIPS/eip-5438.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+2. setSlotUri/getSlotUri call pair used to manage/view uri on slot, and SlotUriUpdated used to announce uri management happens.
+
+### New URI format
+
+This extension tries to add a new URI format, which is prefixed with "hnft://". This URI format enables redirect or connect activity between hNFTs, it enables great imagination.
+
+
+### Interface
+
+#### IERC5438
```
Comment (Pandapip1) [inline EIPS/eip-5438.md:51]:
```suggestion
#### `IERC5438`
```

File: EIPS/eip-5438.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+```
+//SPDX-License-Identifier: Unlicense
```
Comment (Pandapip1) [inline EIPS/eip-5438.md:53]:
```suggestion
// SPDX-License-Identifier: CC0-1.0
```

File: EIPS/eip-5438.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+## Security Considerations
+No security considerations were found.
+
+## Copyright
+Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
+
+## Citation
+Please cite this document as:
+
+Ironman, "EIP-5438: hNFT, ERC-721 Hyperlink Extension," Ethereum Improvement Proposals, no. 5438, August 2022. [Online serial]. Available: https://eips.ethereum.org/EIPS/eip-5438.
```
Comment (Pandapip1) [inline EIPS/eip-5438.md:167]:
```suggestion
```

File: EIPS/eip-5438.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+In addition, new functions introduced in this standard have many similarities with the existing functions in ERC-721. This allows developers to easily adopt the standard quickly.
+
+## Reference Implementation
+You can find an implementation of this standard in [ERC5438.sol](../assets/eip-5438/contracts/ERC5438.sol).
+
+## Security Considerations
+No security considerations were found.
+
+## Copyright
+Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
```
Comment (Pandapip1) [inline EIPS/eip-5438.md:162]:
```suggestion
Copyright and related rights waived via [CC0](../LICENSE.md).
```

File: EIPS/eip-5438.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+### New URI format to enable redirect between hNFTs
+This extension tries to add a new URI format, which is prefixed with `hnft://`. This URI format enables redirect or connect activity between hNFTs, it enables great imagination.
+
+## Backwards Compatibility
+As mentioned in the specifications section, this standard can be fully ERC-721 compatible by adding an extension function set.
+
+In addition, new functions introduced in this standard have many similarities with the existing functions in ERC-721. This allows developers to easily adopt the standard quickly.
+
+## Reference Implementation
+You can find an implementation of this standard in [ERC5438.sol](../assets/eip-5438/contracts/ERC5438.sol).
```
Comment (Pandapip1) [inline EIPS/eip-5438.md:156]:
```suggestion
You can find an implementation of this standard in [`ERC5438.sol`](../assets/eip-5438/contracts/ERC5438.sol).
```

File: EIPS/eip-5438.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+### Authorize slot to address
+We use Address to represent key of slot to ensure it's flexible enough to deal with different use cases. E.g. one can manage slot's URI with an EOA, or manage slot's URI with a contract in which can do more detailed limitation or agreement.
+
+### New URI format to enable redirect between hNFTs
+This extension tries to add a new URI format, which is prefixed with `hnft://`. This URI format enables redirect or connect activity between hNFTs, it enables great imagination.
+
+## Backwards Compatibility
+As mentioned in the specifications section, this standard can be fully ERC-721 compatible by adding an extension function set.
+
+In addition, new functions introduced in this standard have many similarities with the existing functions in ERC-721. This allows developers to easily adopt the standard quickly.
```
Comment (Pandapip1) [inline EIPS/eip-5438.md:153]:
```suggestion
In addition, new functions introduced in this standard have many similarities with the existing functions in EIP-721. This allows developers to easily adopt the standard quickly.
```

File: EIPS/eip-5438.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+We use URI to represent value of slot to ensure it's flexible enough to deal with different use cases. E.g. one can define a custom uri format to indicate a brand new protocol.
+
+### Authorize slot to address
+We use Address to represent key of slot to ensure it's flexible enough to deal with different use cases. E.g. one can manage slot's URI with an EOA, or manage slot's URI with a contract in which can do more detailed limitation or agreement.
+
+### New URI format to enable redirect between hNFTs
+This extension tries to add a new URI format, which is prefixed with `hnft://`. This URI format enables redirect or connect activity between hNFTs, it enables great imagination.
+
+## Backwards Compatibility
+As mentioned in the specifications section, this standard can be fully ERC-721 compatible by adding an extension function set.
```
Comment (Pandapip1) [inline EIPS/eip-5438.md:151]:
```suggestion
As mentioned in the specifications section, this standard can be fully EIP-721 compatible by adding an extension function set.
```

File: EIPS/eip-5438.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+The `SlotAuthorizationCreated` event MUST be emitted when a slot is authorized to an address.
+
+The `SlotAuthorizationRevoked` event MUST be emitted when a slot authorization is revoked.
+
+The `SlotUriUpdated` event MUSt be emitted when a slot's URI is changed.
+
+The `supportInterface` method MUST return true when called with `0x8f65987b`.
+
+### Authentication
+The authorizeSlotTo, revokeAuthorization and revokeAllAuthorizations function is authenticated if and only if the message sender is the owner of the token.
```
Comment (Pandapip1) [inline EIPS/eip-5438.md:137]:
```suggestion
The `authorizeSlotTo`, `revokeAuthorization`, and `revokeAllAuthorizations` functions are authenticated if and only if the message sender is the owner of the token.
```

File: EIPS/eip-5438.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+created: 2022-08-16
+requires: 165, 721
+---
+
+## Abstract
+
+This ERC proposes a new extension of NFT (non-fungible token, aka ERC-721): nft-hyperlink-extention (hNFT), which embedded hyperlinks in NFTs. With hHNFT, the owners of NFTs can authorize a URL slot to a specific address which can be either an externally-owned account (EOA) or a contract account and they are entitled to revoke that authorization as well. Address which got slot authorization, can manage the URI on the slot.
+
+## Motivation
+NFTs have entered mainstream consciousness in a big way as the trading volume spiked 21,000% to [more than $17 billion in 2021](https://www.cnbc.com/2022/03/10/trading-in-nfts-spiked-21000percent-to-top-17-billion-in-2021-report.html). Despite NFTs’ monumental rise as blockchain-based digital assets, we are just starting to scratch the surface of what is possible with NFTs.
```
Comment (Pandapip1) [inline EIPS/eip-5438.md:19]:
```suggestion
NFTs have entered mainstream consciousness in a big way as the trading volume spiked 21,000% to more than $17 billion in 2021. Despite NFTs’ monumental rise as blockchain-based digital assets, we are just starting to scratch the surface of what is possible with NFTs.
```

File: assets/eip-5438/contracts/ERC5438.sol [hunk 12]
```diff
@@ -0,0 +1,130 @@
+//SPDX-License-Identifier: Unlicense
```
Comment (Pandapip1) [inline assets/eip-5438/contracts/ERC5438.sol:1]:
```suggestion
//SPDX-License-Identifier: CC0-1.0
```

File: assets/eip-5438/package.json [hunk 13]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (Pandapip1) [inline assets/eip-5438/package.json:1]: Delete all packaging / workspace files.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

**Comment by abcoathup** [comment]
@Pandapip1 did this EIP get renamed from EIP5489 to EIP5438?

**Comment by coderfengyun** [comment]
> 

Nope, it's has no relation with 5489

File: EIPS/eip-5438.md [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+
+The emergence of NFTs brings more possibilities for creators and Web3 projects to build up decentralized communities, which also shifts the previous company-consumer relationship to more of a partnership. Each holder is entitled to trade, promote, or create based on their NFTs to make profits. 
+
+However, there hasn’t been an effective approach to providing avenues of continued monetization for community holders. Going through all the NFT related EIPs, the solutions to ”making NFTs profitable“ are limited to transfers ([EIP-721](https://eips.ethereum.org/EIPS/eip-721)), rental and mortgage ([EIP 4907](https://eips.ethereum.org/EIPS/eip-4907)).
+
+Inspired by hyperlink, the essence of World Wide Web, hNFT standard is proposed positioning NFT as the medium for Web 3.0. HNFT standard endows NFTs with hyperlink extension so that it is possible to exercise the `fructus` right, meanwhile, keeps the `abusus`. With hyperlink extension, NFT owners can earn profit by authorizing any hyperlink slot to an address.
```
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5438.md:25]: Will do it.

File: EIPS/eip-5438.md [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+eip: 5438
+title: NFT hyperlink extension
+description: hyperlink extension for ERC-721
+author: IronMan_CH (@IronMan_CH)
+discussions-to: https://ethereum-magicians.org/t/eip-5438-NFT-hyperlink-extension/XXX
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-08-16
+requires: 165, 721
```
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5438.md:11]: 4907 is just taken as an example, I think no need to put it into requires.

**Comment by abcoathup** [comment]
> > 
> 
> Nope, it has no relation with 5489

Generally the EIP# is the PR# unless the EIP editor assigns a different number.  The PR is 5489.

**Comment by coderfengyun** [comment]
> 

OK, I'll fix it.

**Comment by coderfengyun** [comment]
> @Pandapip1 did this EIP get renamed from EIP5489 to EIP5438?

Have fixed it.

File: EIPS/eip-5438.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5438.md]: @Pandapip1  I have fixed the grammatical errors. Check it please.

File: EIPS/eip-5489.md [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+
+The emergence of NFTs brings more possibilities for creators and Web3 projects to build up decentralized communities, which also shifts the previous company-consumer relationship to more of a partnership. Each NFT holder is entitled to trade, promote, or create based on their NFTs to make profits. 
+
+However, there hasn’t been an effective approach to providing avenues of continued monetization for community holders. Going through all the NFT-related EIPs, the solutions to "making NFTs profitable" are limited to transfers ([EIP-721](./eip-721.md)), rental and mortgage ([EIP-4907](./eip-4907.md)).
+
+Inspired by hyperlink, the essence of World Wide Web, the proposed hNFT standard positions NFT as the medium for Web 3.0. The hNFT standard endows NFTs with a hyperlink extension so that it is possible to exercise the `fructus` right, meanwhile, keeps the `abusus`. With hyperlink extension, NFT owners can earn profit by authorizing any hyperlink slot to an address.
```
Comment (Pandapip1) [inline EIPS/eip-5489.md:25]:
```suggestion
```
Required: Not motivation.

File: EIPS/eip-5489.md [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+
+NFT hyperlink extension extends EIP-721 with a `mapping<TokenId, mapping<SlotManagerAddress, uri>>`. The mapping means `TokenId` identified NFT has authorized `SlotManagerAddress` to manage the `uri` which is identified by `SlotManagerAddress`.
+
+Besides, extension also provide the supporting facilities, such as
+1. authorizeSlot/revokeSlot call pair used to manage slot authorization, and SlotAuthorizationCreate/Revoked event pair used to announce authorization management happens.
+2. setSlotUri/getSlotUri call pair used to manage/view uri on slot, and SlotUriUpdated used to announce uri management happens.
+
+### New URI format
+
+This extension tries to add a new URI format, which is prefixed with "hnft://". This URI format enables redirect or connect activity between hNFTs, it enables great imagination.
```
Comment (Pandapip1) [inline EIPS/eip-5489.md:45]:
```suggestion
This EIP adds a new URI scheme, "hnft://". This URI scheme enables redirecting or connecting activity between hNFTs.
```
Required: Fix grammar.
Required: Use "this EIP".
Required: Use correct terminology.
Recommended: Rephrasing.

File: EIPS/eip-5489.md [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+
+Right to permission access to information—could be composed with “usus,” “abusus,” and “fructus” to create a nuanced constellation of access rights.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+### NFT hyperlink extension
+
+NFT hyperlink extension extends EIP-721 with a `mapping<TokenId, mapping<SlotManagerAddress, uri>>`. The mapping means `TokenId` identified NFT has authorized `SlotManagerAddress` to manage the `uri` which is identified by `SlotManagerAddress`.
```
Comment (Pandapip1) [inline EIPS/eip-5489.md:37]:
```suggestion
This EIP extends EIP-721 with a `mapping<TokenId, mapping<SlotManagerAddress, uri>>`. The mapping means `TokenId` identified NFT has authorized `SlotManagerAddress` to manage the `uri` which is identified by `SlotManagerAddress`.
```
Required: Use "this EIP".

File: EIPS/eip-5489.md [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+
+The `supportInterface` method MUST return true when called with `0x8f65987b`.
+
+### Authentication
+The `authorizeSlotTo`, `revokeAuthorization`, and `revokeAllAuthorizations` functions are authenticated if and only if the message sender is the owner of the token.
+
+## Rationale
+
+### Extends NFT with hyperlinks
+We use URI to represent value of slot to ensure it's flexible enough to deal with different use cases. E.g. one can define a custom uri format to indicate a brand new protocol.
```
Comment (Pandapip1) [inline EIPS/eip-5489.md:140]:
```suggestion
URIs are used to represent the value of slots to ensure enough flexibility to deal with different use cases.
```
Required: Fix grammar.
Recommended: Rephrasing.

File: EIPS/eip-5489.md [hunk 20]
```diff
@@ -0,0 +1,160 @@
...
+
+### Authorize slot to address
+We use Address to represent key of slot to ensure it's flexible enough to deal with different use cases. E.g. one can manage slot's URI with an EOA, or manage slot's URI with a contract in which can do more detailed limitation or agreement.
```
Comment (Pandapip1) [inline EIPS/eip-5489.md:143]:
```suggestion
We use addresses to represent the key of slots to ensure enough flexibility to deal with all use cases.
```
Required: Fix grammar.
Recommended: Rephrasing.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

File: EIPS/eip-5489.md (hunk 16 above)
Comment (coderfengyun) [reply EIPS/eip-5489.md:25]: @Pandapip1 How about just keeping "With hyperlink extension, NFT owners can earn profit by authorizing any hyperlink slot to an address." this sentence ?

Comment (Pandapip1) [reply EIPS/eip-5489.md:25]: Again, that describes the solution, not the problem, and so would belong in the Abstract.

Comment (coderfengyun) [reply EIPS/eip-5489.md:25]: I see, the core problems are:
1. no on-chain way to take advantage of NFT's influence without affecting ownership for now.
2. no standard on how to use in URI in NFT relative standard

Comment (coderfengyun) [reply EIPS/eip-5489.md:25]: @Pandapip1 How about just list these two as the motivation?

File: EIPS/eip-5489.md [hunk 21]
```diff
@@ -0,0 +1,160 @@
+---
+eip: 5489
+title: NFT hyperlink extension
+description: hyperlink extension for EIP-721
```
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md:4]: The meaning can be extended by usages. For example, NFTs can be taken as billboards.

File: EIPS/eip-5489.md [hunk 22]
```diff
@@ -0,0 +1,160 @@
...
+---
+
+## Abstract
+
+This EIP proposes a new extension for NFTs (non-fungible token, aka [EIP-721](./eip-721.md)): nft-hyperlink-extention (hNFT), embedding NFTs with hyperlinks, referred to as “hNFTs”. As owners of hNFTs, users may authorize a URL slot to a specific address which can be either an externally-owned account (EOA) or a contract address and hNFT owners are entitled to revoke that authorization at any time. The address which has slot authorization can manage the URI of that slot.
+
+## Motivation
+NFTs have entered mainstream consciousness in a big way as NFT trading volume spiked 21,000% to more than $17 billion in 2021. Despite NFTs’ monumental rise as blockchain-based digital assets, we are just starting to scratch the surface of what is possible with NFTs.
+
+The emergence of NFTs brings more possibilities for creators and Web3 projects to build up decentralized communities, which also shifts the previous company-consumer relationship to more of a partnership. Each NFT holder is entitled to trade, promote, or create based on their NFTs to make profits.
```
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md:21]: We rephrase this paragraph.

File: EIPS/eip-5489.md (hunk 16 above)
Comment (coderfengyun) [reply EIPS/eip-5489.md]: @Pandapip1 I've modified the motivation.

File: EIPS/eip-5489.md [hunk 23]
```diff
@@ -0,0 +1,160 @@
...
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-08-16
+requires: 165, 721
+---
+
+## Abstract
+
+This EIP proposes a new extension for NFTs (non-fungible token, aka [EIP-721](./eip-721.md)): nft-hyperlink-extention (hNFT), embedding NFTs with hyperlinks, referred to as “hNFTs”. As owners of hNFTs, users may authorize a URL slot to a specific address which can be either an externally-owned account (EOA) or a contract address and hNFT owners are entitled to revoke that authorization at any time. The address which has slot authorization can manage the URI of that slot.
```
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md]: In my original thought, It's URL slot.

File: EIPS/eip-5489.md (hunk 22 above)
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md]: @SamWilsn I've rephrase motivation part.

File: EIPS/eip-5489.md [hunk 24]
```diff
@@ -0,0 +1,160 @@
...
+
+However, there hasn’t been an effective approach to providing avenues of continued monetization for community holders. Going through all the NFT-related EIPs, the solutions to "making NFTs profitable" are limited to transfers ([EIP-721](./eip-721.md)), rental and mortgage ([EIP-4907](./eip-4907.md)).
+
+Inspired by hyperlink, the essence of World Wide Web, the proposed hNFT standard positions NFT as the medium for Web 3.0. The hNFT standard endows NFTs with a hyperlink extension so that it is possible to exercise the `fructus` right, meanwhile, keeps the `abusus`. With hyperlink extension, NFT owners can earn profit by authorizing any hyperlink slot to an address.
+
+To establish standardization for NFT hyperlinks, it is important to share a unified standard so as to facilitate collaboration among all applications.
+
+Right to permission access to information—could be composed with “usus,” “abusus,” and “fructus” to create a nuanced constellation of access rights.
```
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md]: This  paragraph is removed.

File: EIPS/eip-5489.md [hunk 25]
```diff
@@ -0,0 +1,160 @@
...
+    /**
+     * @dev this event emits when the authorization on slot `slotManagerAddr` of token `tokenId` is revoked.
+     * So, the corresponding DApp can handle this to stop on-going incentives or rights
+     */
+    event SlotAuthorizationRevoked(uint256 tokenId, address slotManagerAddr);
+
+    /**
+     * @dev this event emits when the uri on slot `slotManagerAddr` of token `tokenId` has been updated to `uri`.
+     */
+    event SlotUriUpdated(uint256 tokenId, address slotManagerAddr, string uri);
```
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md:61]: Will fix it.

File: EIPS/eip-5489.md [hunk 26]
```diff
@@ -0,0 +1,160 @@
...
+    function setSlotUri(
+        uint256 tokenId,
+        string calldata newUri
+    ) external;
+
+    /**
+     * @dev
+     * returns the latest uri of an slot on a token, which is indicated by `tokenId`, `slotManagerAddr`
+     */
+    function getSlotUri(uint256 tokenId, address slotManagerAddr)
```
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md:101]: Do you mean [eip-3525](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-3525.md) ?

File: EIPS/eip-5489.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md]: I've rephrase this description.

File: EIPS/eip-5489.md (hunk 23 above)
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md]: I've add usage in advertisement industry.

File: EIPS/eip-5489.md (hunk 25 above)
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md]: I've fixed this.

File: EIPS/eip-5489.md [hunk 27]
```diff
@@ -0,0 +1,160 @@
...
+
+Right to permission access to information—could be composed with “usus,” “abusus,” and “fructus” to create a nuanced constellation of access rights.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+### NFT hyperlink extension
+
+This EIP extends EIP-721 with a `mapping<TokenId, mapping<SlotManagerAddress, uri>>`. The mapping means `TokenId` identified NFT has authorized `SlotManagerAddress` to manage the `uri` which is identified by `SlotManagerAddress`.
```
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md]: Fixed

File: EIPS/eip-5489.md [hunk 28]
```diff
@@ -0,0 +1,160 @@
...
+
+Besides, extension also provide the supporting facilities, such as
+1. authorizeSlot/revokeSlot call pair used to manage slot authorization, and SlotAuthorizationCreate/Revoked event pair used to announce authorization management happens.
+2. setSlotUri/getSlotUri call pair used to manage/view uri on slot, and SlotUriUpdated used to announce uri management happens.
```
[the remark this answers is not in the document]
Comment (coderfengyun) [reply EIPS/eip-5489.md]: Fixed.

**Review by Pandapip1 — APPROVED** [review]
This is fine. There are some things that would be good to fix, but they aren't blockers.

**Pull request merged**